### ISSUE TITLE
check_variables false positive

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#454]
FIXED,

the INIT part is a variable definition?

```abap
REDUCE string( INIT result = ||
                                FOR <result> IN it_result
                                NEXT result = result && _get_html_for_token( i_result = <result>
                                                                             i_match  = i_match ) )
```

**Comment by larshp** [comment]
add `icon_modify`

**Comment by larshp** [comment]
and LET, 

```abap
cl_abap_structdescr=>create(
                        VALUE #( LET type = CAST cl_abap_datadescr( cl_abap_typedescr=>describe_by_name( |ICON_D| ) )
                                 IN
                                 FOR x = 1 WHILE x <= board->m_size
                                 ( name = |COL_{ x }|
                                   type = type ) ) ).
```

**Comment by larshp** [comment]
alive is a component, not a source,

```abap
    r_cells_alive = lines( FILTER #( cells USING KEY key_alive
                                           WHERE alive = abap_true  ) ).
```

**Comment by larshp** [comment]
`CREATE DATA ls_out-table TYPE string.`

**Comment by larshp** [comment]
```abap
DATA(result) = REDUCE string( INIT result = ||
        FOR <result> IN it_result
        NEXT result = result && 'abc' ).
```

**Comment by larshp** [comment]
```typescript
  it("REDUCE with INIT", () => {
    const abap = "DATA it_result TYPE c.\n" +
      "DATA(output) = REDUCE string( INIT result = ||\n" +
      "  FOR <result> IN it_result\n" +
      "  NEXT result = result && 'abc' ).";
    const issues = runProgram(abap);
    console.dir(issues);
    expect(issues.length).to.equals(0);
  });
```
